corehq/apps/userreports/tasks.py

```diff
@@ -178,25 +186,52 @@ def _run_report(backend_to_use):
     return objects
 
 
+@periodic_task(
+    run_every=ASYNC_INDICATOR_QUEUE_TIME,
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def queue_async_indicators():
+    start = datetime.utcnow()
+    cutoff = start + ASYNC_INDICATOR_QUEUE_TIME
+    time_for_crit_section = ASYNC_INDICATOR_QUEUE_TIME.seconds - 10
+
+    with CriticalSection(['queue-async-indicators'], timeout=time_for_crit_section):
+        redis_client = get_redis_client().client.get_client()
+        indicators = AsyncIndicator.objects.all()[:10000]
+        paginator = Paginator(indicators, 1000)
+        for page in paginator.page_range:
+            for indicator in paginator.page(page).object_list:
+                now = datetime.utcnow()
+                if now > cutoff:
```

Comment:
I don't think it's very easy to do that because we use a deprecated decorator https://github.com/celery/celery/blob/3.1/celery/task/base.py#L171-L179

would be good to see how easy it is to upgrade to the recommended way to make tasks. not sure how hard that will be though